pkg/authenticator/authenticator.go

```diff
@@ -219,12 +247,19 @@ func (auth *Authenticator) Authenticate() ([]byte, error) {
 		log.Debug(log.CAKC038)
 
 		if err := auth.Login(); err != nil {
-			return nil, err
+			return err
 		}
 
 		log.Debug(log.CAKC037)
 	}
 
+	return nil
+}
+
+// makeAuthenticationRequest reads the cert from memory and uses it to send
+// an authentication request to the Conjur server. It also validates the response
+// code before returning its body
+func (auth *Authenticator) makeAuthenticationRequest() ([]byte, error) {
```

Comment:
Good method separation!
IMO `sendAuthenticationRequest` is better than `makeAuthenticationRequest`
```suggestion
func (auth *Authenticator) sendAuthenticationRequest() ([]byte, error) {
```